onnxruntime/python/tools/kernel_explorer/kernels/gemm_fast_gelu_test.py

```diff
@@ -0,0 +1,197 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import re
+import sys
+from itertools import product
+
+import kernel_explorer as ke
+import numpy as np
+import pytest
+from utils import get_gemm_bound
+
+
+def transab_to_suffix(transab):
+    return {
+        (True, True): "TT",
+        (True, False): "TN",
+        (False, True): "NT",
+        (False, False): "NN",
+    }[tuple(transab)]
+
+
+def dtype_to_funcs(dtype):
+    type_map = {
+        "float16": list(filter(lambda x: re.search("GemmFastGelu.*_half", x), dir(ke))),
+        "float32": list(filter(lambda x: re.search("GemmFastGelu.*_float", x), dir(ke))),
+    }
+    return type_map[dtype]
+
+
+def fast_gelu(x, bias):
+    x = x + bias
+    y = 0.5 * x * (1 + np.tanh(0.797885 * x + 0.035677 * x * x * x))
+    return y
+
+
+# TODO The test method needs update.
+def _test_gemmfastgelu(func, dtype: str, m: int, n: int, k: int, transa=False, transb=False):
+    assert dtype in ["float16", "float32"]
+
+    a_shape = (k, m) if transa else (m, k)
+    b_shape = (n, k) if transb else (k, n)
+
+    np.random.seed(0)
+    a = (np.random.rand(*a_shape)).astype(dtype).astype("float64")
+    b = (np.random.rand(*b_shape)).astype(dtype).astype("float64")
+    bias = (np.random.rand(n)).astype(dtype)
+    temp_c = (a.T if transa else a) @ (b.T if transb else b)
+
+    bound = get_gemm_bound(dtype, a, b, temp_c, transa, transb)
+
+    temp_c = temp_c.astype(dtype)
+    ref_c = fast_gelu(temp_c, bias)
+
+    a = a.astype(dtype)
+    b = b.astype(dtype)
+
+    my_c = np.zeros((m, n), dtype=dtype)
+    dev_a = ke.DeviceArray(a)
+    dev_b = ke.DeviceArray(b)
+    dev_bias = ke.DeviceArray(bias)
+    dev_c = ke.DeviceArray(my_c)
+
+    opa = ke.blas_op.T if transa else ke.blas_op.N
+    opb = ke.blas_op.T if transb else ke.blas_op.N
+    lda = a_shape[1]
+    ldb = b_shape[1]
+    alpha = 1.0
+    beta = 0.0
+    my_func = getattr(ke, func)
+    my_op = my_func(opa, opb, m, n, k, alpha, dev_a, lda, dev_b, ldb, dev_bias, beta, dev_c, n)
+
+    if my_op.IsSupported():
+        my_op.Run()
+        dev_c.UpdateHostNumpyArray()
+
+        print(
+            f"{func:<50} : dtype={dtype} {transab_to_suffix((transa, transb))} m={m:<5} n={n:<5} k={k:<5} bound: {bound}"
+        )
+        try:
+            np.testing.assert_allclose(my_c, ref_c, rtol=max(bound, 1e-2))
+        except Exception as err:
+            header = "*" * 30 + func + "*" * 30
+            print(header)
+            print(err)
+            print("*" * len(header))
+            raise Exception(str(err))
```

Comment:
This part can be simplified to `np.testing.assert_allclose(my_c, ref_c, rtol=max(bound, 1e-2))`